dbt/include/athena/macros/materializations/snapshots/snapshot.sql

```diff
@@ -205,7 +195,7 @@
         {% for column in source_columns %}
           {% if column.name == 'dbt_valid_to' %}
           CASE
-          WHEN dbt_valid_to=end_of_time() AND is_current_record
+          WHEN dbt_valid_to=CAST('9999-12-31' AS timestamp) AND is_current_record
```

Comment:
Based on the athena engine in use this might break I guess. Maybe is better to wrap the cast in a macro that the final User can control. Wdyt?

Reply:
I tried that in a previous commit but I kept getting function end_of_time not registered. This is the macro I added

```
{% macro athena__end_of_time() -%}
    cast('9999-12-31' as timestamp)
{%- endmacro %}
```

Any idea what I'm doing wrong? I also tried referencing the function as adapter.end_of_time() which also didn't seem to work.

Reply:
Not sure if matter, but there is a missing `-` in your code.
```
{%- macro athena__end_of_time() -%}
    cast('9999-12-31' as timestamp)
{%- endmacro %}
```

Reply:
I have extracted this into a function `end_of_time`. I initially fumbled my function call without surrounding {{ }}